### PR TITLE
 Refactor openshift.common.deployment_type

### PR DESCRIPTION
**Pull request by michaelgugino**
This commit refactors openshift.common.deployment_type
and openshift.common.deployment_subtype.

This commit is based on openshift.common.version refactor PR.

### CODE REVIEW

File: roles/openshift_facts/library/openshift_facts.py [hunk 1]
```diff
@@ -519,83 +491,6 @@ def set_aggregate_facts(facts):
...
-            facts (dict): existing facts
-        Returns:
-            dict: the facts dict updated with version facts.
-    """
-    if 'common' in facts:
-        openshift_version = get_openshift_version(facts)
-        if openshift_version and openshift_version != "latest":
-            version = LooseVersion(openshift_version)
-            facts['common']['version'] = openshift_version
-            facts['common']['short_version'] = '.'.join([str(x) for x in version.version[0:2]])
```
Comment (vrutkovs) [inline roles/openshift_facts/library/openshift_facts.py]: Some places still use `short_version`:

```
TASK [Update registry-console image to current version] ***************************************************************************************************************************************
task path: /home/vrutkovs/workdir/tmp.3zoyyI3KYA/openshift-ansible/playbooks/common/openshift-cluster/upgrades/post_control_plane.yml:84
Tuesday 06 March 2018  14:22:44 +0100 (0:00:01.610)       0:19:33.476 ********* 
fatal: [vrutkovs_tmp.3zoyyI3KYA-master-1]: FAILED! => {
    "msg": "The task includes an option with an undefined variable. The error was: {{ openshift.master.registry_url | regex_replace ( '(origin|ose)-\\${component}', 'registry-console') | replace ( '${version}', 'v' ~ openshift.common.short_version ) }}: 'dict object' has no attribute 'short_version'\n\nThe error appears to have been in '/home/vrutkovs/workdir/tmp.3zoyyI3KYA/openshift-ansible/playbooks/common/openshift-cluster/upgrades/post_control_plane.yml': line 84, column 5, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n  - name: Update registry-console image to current version\n    ^ here\n\nexception type: <class 'ansible.errors.AnsibleUndefinedVariable'>\nexception: {{ openshift.master.registry_url | regex_replace ( '(origin|ose)-\\${component}', 'registry-console') | replace ( '${version}', 'v' ~ openshift.common.short_version ) }}: 'dict object' has no attribute 'short_version'"
}
```

Comment (vrutkovs) [reply roles/openshift_facts/library/openshift_facts.py]: https://github.com/vrutkovs/openshift-ansible/commit/a594c3ba1b3ba4f4ec8e162aace86781cfd3f0d7 fixes it

Comment (michaelgugino) [reply roles/openshift_facts/library/openshift_facts.py]: short_version will be removed from openshift_hosted in my other patch, which should merge before this one.  Removing it here will just make me rebase ;)

Comment (vrutkovs) [reply roles/openshift_facts/library/openshift_facts.py]: Right, it works with all PRs applied

**Comment by michaelgugino** [comment]
/hold

I will rebase and squash after the version patch merges.

**Comment by vrutkovs** [comment]
Fresh deploy and upgrade from 3.7 worked fine

/lgtm

**Pull request merged**